compaction/compaction_strategy.cc

```diff
@@ -177,25 +177,35 @@ double size_tiered_backlog_tracker::backlog(const compaction_backlog_tracker::on
     return b > 0 ? b : 0;
 }
 
-// FIXME: Should provide strong exception safety guarantees
+// Provides strong exception safety guarantees.
 void size_tiered_backlog_tracker::replace_sstables(const std::vector<sstables::shared_sstable>& old_ssts, const std::vector<sstables::shared_sstable>& new_ssts) {
+    auto all = _all;
+    auto total_bytes = _total_bytes;
+    all.reserve(all.size() + new_ssts.size());
```

Comment:
This can cause a thousand allocations with 1TB/shard and 1GB/sstable. A thousand allocations isn't quite a stall, but it isn't great either.

Please see if we can switch to absl::flat_hash_set or node_hash_set.

Reply:
(post-merge)